exporter/loadbalancingexporter/resolver.go

```diff
@@ -0,0 +1,33 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loadbalancingexporter
+
+import "context"
+
+// resolver determines the contract for sources of backend endpoint information
+type resolver interface {
+	// resolve returns the current list of endpoints.
+	// returns either a non-nil error and a nil list of endpoints, or a non-nil list of endpoints and nil error.
+	resolve(context.Context) ([]string, error)
+
+	// start signals the resolver to start its work
+	start(context.Context) error
+
+	// shutdown signals the resolver to finish its work
+	shutdown(context.Context) error
+
+	// onChange registers a function to call back whenever the list of backends is updated
```

Comment:
1. Can this be called before `start`?
2. If it is called after `start` is it expected that the callback will be called immediately or soon after with the current list of backends?

Reply:
I added a comment making it clear that `onChange` has to be called before `start`, otherwise the first update might be missed.

So:

1. yes, in fact, it *should* be called before start (not "must", but "should")
1. no, this is only called when the resolver gets a different set of backends